src/coreclr/tools/ILVerify/Program.cs

```diff
@@ -351,6 +347,33 @@ private void VerifyTypes(PEReader peReader, EcmaModule module, string path, ref
             }
         }
 
+        /// <summary>
+        /// Returns full class name, includes parent class for nested class.
+        /// </summary>
+        private string GetFullClassName(MetadataReader metadataReader, TypeDefinitionHandle typeDefinitionHandle)
+        {
+            var typeDef = metadataReader.GetTypeDefinition(typeDefinitionHandle);
+            var typeName = metadataReader.GetString(typeDef.Name);
+
+            var fullName = new StringBuilder();
+            var namespaceName = metadataReader.GetString(typeDef.Namespace);
+
+            var declaringType = typeDef.GetDeclaringType();
+            if (string.IsNullOrEmpty(namespaceName) && !declaringType.IsNil)
+            {
+                fullName.Append(GetFullClassName(metadataReader, declaringType));
+                fullName.Append("+");
+            }
+            else
+            {
+                fullName.Append(namespaceName);
+                fullName.Append(".");
+            }
```

Comment:
```suggestion
            if (!declaringType.IsNil)
            {
                fullName.Append(GetFullClassName(metadataReader, declaringType));
                fullName.Append('+');
            }
            
            if (!string.IsNullOrEmpty(namespaceName))
            {
                fullName.Append(namespaceName);
                fullName.Append('.');
            }
```
The nested types can have namespace as well. (C# does not allow it, but it is possible in IL.)